src/stream-analytics/azext_stream_analytics/vendored_sdks/streamanalytics/models/_models.py

```diff
@@ -0,0 +1,2455 @@
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+
+from msrest.serialization import Model
+
+
+class Serialization(Model):
+    """Describes how data from an input is serialized or how data is serialized
+    when written to an output.
+
+    You probably want to use the sub-classes and not this class directly. Known
+    sub-classes are: AvroSerialization, JsonSerialization, CsvSerialization
+
+    All required parameters must be populated in order to send to Azure.
+
+    :param type: Required. Constant filled by server.
+    :type type: str
+    """
+
+    _validation = {
+        'type': {'required': True},
+    }
+
+    _attribute_map = {
+        'type': {'key': 'type', 'type': 'str'},
+    }
+
+    _subtype_map = {
+        'type': {'Avro': 'AvroSerialization', 'Json': 'JsonSerialization', 'Csv': 'CsvSerialization'}
+    }
+
+    def __init__(self, **kwargs):
+        super(Serialization, self).__init__(**kwargs)
+        self.type = None
+
+
+class AvroSerialization(Serialization):
+    """Describes how data from an input is serialized or how data is serialized
+    when written to an output in Avro format.
+
+    All required parameters must be populated in order to send to Azure.
+
+    :param type: Required. Constant filled by server.
+    :type type: str
+    :param properties: The properties that are associated with the Avro
+     serialization type. Required on PUT (CreateOrReplace) requests.
+    :type properties: object
+    """
+
+    _validation = {
+        'type': {'required': True},
+    }
+
+    _attribute_map = {
+        'type': {'key': 'type', 'type': 'str'},
+        'properties': {'key': 'properties', 'type': 'object'},
+    }
+
+    def __init__(self, **kwargs):
+        super(AvroSerialization, self).__init__(**kwargs)
+        self.properties = kwargs.get('properties', None)
+        self.type = 'Avro'
+
+
+class OutputDataSource(Model):
+    """Describes the data source that output will be written to.
+
+    You probably want to use the sub-classes and not this class directly. Known
+    sub-classes are: AzureDataLakeStoreOutputDataSource,
+    PowerBIOutputDataSource, ServiceBusTopicOutputDataSource,
+    ServiceBusQueueOutputDataSource, DocumentDbOutputDataSource,
+    AzureSqlDatabaseOutputDataSource, EventHubOutputDataSource,
+    AzureTableOutputDataSource, BlobOutputDataSource
+
+    All required parameters must be populated in order to send to Azure.
+
+    :param type: Required. Constant filled by server.
+    :type type: str
+    """
+
+    _validation = {
+        'type': {'required': True},
+    }
+
+    _attribute_map = {
+        'type': {'key': 'type', 'type': 'str'},
+    }
+
+    _subtype_map = {
```

Comment:
well. It's quite complicate.

Reply:
I agree, for such a complicate definition, we can support json input. Is there any way to let user know how to use it?

Reply:
yes, it's complicate for these parameters. Actually there are some examples in the rest spec examples, can we link it directly, or write them down in our README file or CLI examples (around 10 examples for each)?

Reply:
On the one hand, I totally agree that it's quite complicated and it would take too much effort to support without json argument. On the other hand, for such a complicated argument, it's quite hard to use a json argument for users. examples is good. Is it able to add some commands like `az vm diagnostics get-default-config -h` to give examples to users. Meanwhile add some simple checks and hints for users?

Reply:
@MyronFanQiu I add more json format examples for `input create`, `output create` and `function create`. Not cover all the cases, but most of them are covered.